spec/api/keys/keys_spec.rb

```diff
@@ -376,6 +388,38 @@
     end
   end
 
+  context "when a user's default key has an expiration date" do
+    before(:each) do
+      system("chef-server-ctl delete-user-key #{user['name']} default")
+      system("chef-server-ctl add-user-key #{user['name']} #{$public_key_filepath} --key-name default --expiration-date 2017-12-24T21:00:00")
+    end
+    context "when a user's default key is updated via a PUT to /users/:user" do
+      before(:each) do
+        original_data = JSON.parse(get("#{platform.server}/users/#{user['name']}", superuser))
+        put("#{platform.server}/users/#{user['name']}", superuser, :payload => JSON.generate(original_data))
```

Comment:
I'm still testing locally, but I'm 99% sure that you need to modify original_data before PUTing it back:

https://github.com/chef/oc_erchef/blob/master/apps/oc_chef_wm/src/oc_chef_wm_base.erl#L890-L895

We do this tricky thing were we don't actually write to the database if you update the object but don't actually change any data.  Since this all works because of a trigger, no update -> no trigger.